activerecord/test/cases/transactions_test.rb

```diff
@@ -18,6 +18,10 @@ def setup
     @first, @second = Topic.find(1, 2).sort_by(&:id)
   end
 
+  teardown do
+    ActiveRecord::Base.clear_all_connections!
```

Comment:
Why do we need to clear the connections here?

Reply:
If one of the tests added in #40541 fail, hundreds of unrelated tests fail because the DB is left in an inconsistent state. Resetting the connections preemptively during teardown limits those unrelated failures so that any problem can be much more easily debugged. Can remove if this is an overreach.

Reply:
Thanks for highlighting the issue with the `sqlite3_mem` suite. I've been using rails-dev-box which doesn't seem to run that suite, so missed the failures.

I've changed this to an ensure clause on just the two related tests. Let me know if you had a different pattern in mind though.